src/internal/m365/collection/drive/delta_tree.go

```diff
@@ -468,11 +468,15 @@ func walkTreeAndBuildCollections(
 				"path_suffix", parentPath.Elements())
 	}
 
-	// add the folder itself to the list of files inside the folder.
-	// that will cause the collection processor to generate a metadata
-	// file to hold the folder's permissions.
-	files := maps.Clone(node.files)
-	files[id] = node.folder
+	files := node.files
+
+	if !isRoot {
```

Comment:
does adding the root cause failures with Graph or is this to be more inline with the existing implementation? Not having a special case for this would be a bit cleaner if it doesn't have adverse effects and isn't difficult to get test code setup for